cli/src/scheduler/helpers.rs

```diff
@@ -0,0 +1,467 @@
+use {
+    crate::{prelude::*, workflow},
+    nexus_sdk::{
+        crypto::session::Session,
+        idents::{primitives, sui_framework, workflow as workflow_idents},
+        sui::{self, move_ident_str, MoveStructTag, MoveTypeTag},
+        transactions::scheduler as scheduler_tx,
+        types::{NexusObjects, DEFAULT_ENTRY_GROUP},
+    },
+    serde_json::Value,
+    std::collections::HashMap,
+};
+
+/// Parse metadata pairs provided as `key=value` strings.
+pub(crate) fn parse_metadata(pairs: &[String]) -> AnyResult<Vec<(String, String)>, NexusCliError> {
+    let mut result = Vec::with_capacity(pairs.len());
+    for pair in pairs {
+        let Some((key, value)) = pair.split_once('=') else {
+            return Err(NexusCliError::Any(anyhow!(
+                "Invalid metadata entry '{pair}'. Expected format key=value"
+            )));
+        };
+        if key.trim().is_empty() {
+            return Err(NexusCliError::Any(anyhow!(
+                "Metadata key in '{pair}' cannot be empty"
+            )));
+        }
+        result.push((key.trim().to_owned(), value.trim().to_owned()));
+    }
+    Ok(result)
+}
+
+pub(crate) fn metadata_argument(
+    tx: &mut sui::ProgrammableTransactionBuilder,
+    objects: &NexusObjects,
+    metadata: &[(String, String)],
+) -> AnyResult<sui::Argument, NexusCliError> {
+    scheduler_tx::new_metadata(tx, objects, metadata.iter().cloned())
+        .map_err(|e| NexusCliError::Any(anyhow!(e)))
+}
+
+pub(crate) fn constraints_policy_argument(
+    tx: &mut sui::ProgrammableTransactionBuilder,
+    objects: &NexusObjects,
+) -> AnyResult<sui::Argument, NexusCliError> {
+    let symbol_type = policy_symbol_type_tag(objects.primitives_pkg_id);
+    let time_constraint_tag = time_constraint_type_tag(objects.workflow_pkg_id);
+
+    let constraint_symbol = tx.programmable_move_call(
+        objects.primitives_pkg_id,
+        move_ident_str!("policy").into(),
+        move_ident_str!("witness_symbol").into(),
+        vec![time_constraint_tag],
+        vec![],
+    );
+
+    let constraint_sequence = tx.programmable_move_call(
+        sui::MOVE_STDLIB_PACKAGE_ID,
+        move_ident_str!("vector").into(),
+        move_ident_str!("singleton").into(),
+        vec![symbol_type.clone()],
+        vec![constraint_symbol],
+    );
+
+    let constraints = tx.programmable_move_call(
+        objects.workflow_pkg_id,
+        move_ident_str!("scheduler").into(),
+        move_ident_str!("new_constraints_policy").into(),
+        vec![],
+        vec![constraint_sequence],
+    );
+
+    let config = scheduler_tx::new_time_constraint_config(tx, objects)
+        .map_err(|e| NexusCliError::Any(anyhow!(e)))?;
+    scheduler_tx::register_time_constraint(tx, objects, constraints.clone(), config)
+        .map_err(|e| NexusCliError::Any(anyhow!(e)))?;
+
+    Ok(constraints)
+}
```

Comment:
All functions that take `tx: &mut sui::ProgrammableTransactionBuilder` should be in `nexus-sdk::transactions` module so that anyone can reuse these tx templates